src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,240 @@
+                    return;
+                }
+                investedAssetsAmount = 0;
+            } else {
+                investedAssetsAmount -= balancesDiff;
+            }
+            yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(address(yieldVault), balancesDiff, 0);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        YieldParams storage params = yieldParams;
+
+        address yieldAddress = params.yield;
+
+        if (yieldAddress == address(0)) {
+            return 0;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender, "ZkBobCompoundingPool: Claim is an operator-called method"
+        );
+
+        uint256 dust = params.dust;
+        minClaimAmount = minClaimAmount > dust ? minClaimAmount : dust;
+
+        return _claim(minClaimAmount, yieldAddress, params.interestReceiver, dust);
+    }
+
+    function emergencyWithdraw() external onlyOwner {
+        YieldParams storage params = yieldParams;
+
+        address yieldAddress = params.yield;
+
+        if (yieldAddress == address(0)) {
+            return;
+        }
+
+        IERC4626 yieldVault = IERC4626(yieldAddress);
+
+        uint256 currentInvestedSharesAmount = yieldVault.balanceOf(address(this));
+        uint256 allAssets = yieldVault.convertToAssets(currentInvestedSharesAmount);
+        uint256 currentInvestedAssetsAmount = investedAssetsAmount;
+        uint256 rebalancedAssets = allAssets;
+
+        yieldVault.withdraw(allAssets, address(this), address(this));
+        if (allAssets > currentInvestedAssetsAmount) {
+            IERC20(token).transfer(address(params.interestReceiver), allAssets - currentInvestedAssetsAmount);
+            rebalancedAssets = currentInvestedAssetsAmount;
+            emit Claimed(yieldAddress, allAssets - currentInvestedAssetsAmount);
+        }
+        investedAssetsAmount = 0;
+        emit Rebalance(yieldAddress, rebalancedAssets, 0);
+    }
+
+    function _claim(
+        uint256 minClaimAmount,
+        address yieldAddress,
+        address interestReceiver,
+        uint256 dust
+    )
+        internal
+        returns (uint256)
+    {
+        IERC4626 yieldVault = IERC4626(yieldAddress);
+        uint256 currentInvestedSharesAmount = yieldVault.balanceOf(address(this));
+        uint256 lockedAmount = investedAssetsAmount + dust;
+        uint256 allAssets = yieldVault.convertToAssets(currentInvestedSharesAmount);
+
+        if (allAssets < lockedAmount) {
+            return 0;
+        }
+
+        uint256 toClaimAmount = allAssets - investedAssetsAmount;
```

Comment:
The `dust` parameter is a "part of accumulated interest that is non-claimable". It should not be claimable unless the yield is turned off completely.